mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapperTest.java

```diff
@@ -69,22 +82,239 @@
-                CassandraConfiguration.DEFAULT_CONFIGURATION));
+            aclMapper);
+    }
+
+    @Nested
+    class ConsistencyTest {
+
+        private CassandraId inboxId;
+        private MailboxPath inboxPath;
+        private Mailbox inbox;
+        private MailboxPath inboxPathRenamed;
+        private Mailbox inboxRenamed;
+        private MailboxQuery.UserBound inboxSearchQuery;
+        private MailboxQuery.UserBound inboxRenamedSearchQuery;
+
+        @BeforeEach
+        void setUp() {
+            inboxId = CassandraId.timeBased();
+            inboxPath = MailboxPath.forUser(USER, INBOX);
+            inbox = new Mailbox(inboxPath, UID_VALIDITY, inboxId);
+
+            inboxPathRenamed = MailboxPath.forUser(USER, INBOX_RENAMED);
+            inboxRenamed = new Mailbox(inboxPathRenamed, UID_VALIDITY, inboxId);
+            inboxSearchQuery = MailboxQuery.builder()
+                .userAndNamespaceFrom(inboxPath)
+                .expression(new ExactName(INBOX))
+                .build()
+                .asUserBound();
+            inboxRenamedSearchQuery = MailboxQuery.builder()
+                .userAndNamespaceFrom(inboxPathRenamed)
+                .expression(new ExactName(INBOX_RENAMED))
+                .build()
+                .asUserBound();
+        }
+
+        @Test
+        void saveOnCreateShouldBeConsistentWhenFailToPersistMailbox() {
+            doReturn(Mono.error(new RuntimeException("mock exception")))
+                .when(mailboxDAO)
+                .save(inbox);
+
+            doQuietly(() -> testee.save(inbox));
+
+            SoftAssertions.assertSoftly(softly -> {
+                softly.assertThatThrownBy(() -> testee.findMailboxById(inboxId))
+                    .isInstanceOf(MailboxNotFoundException.class);
+                softly.assertThatThrownBy(() -> testee.findMailboxByPath(inboxPath))
+                    .isInstanceOf(MailboxNotFoundException.class);
+                softly.assertThat(testee.findMailboxWithPathLike(inboxSearchQuery))
+                    .isEmpty();
+            });
+        }
+
+        @Test
+        void saveOnRenameThenFailToGetACLShouldBeConsistentWhenFindByInbox() throws Exception {
+            testee.save(inbox);
+
+            when(aclMapper.getACL(inboxId))
+                .thenReturn(Mono.error(new RuntimeException("mock exception")))
+                .thenCallRealMethod();
+
+            doQuietly(() -> testee.save(inboxRenamed));
+
+            SoftAssertions.assertSoftly(Throwing.consumer(softly -> {
+                softly.assertThat(testee.findMailboxById(inboxId))
+                    .isEqualTo(inbox);
+                softly.assertThat(testee.findMailboxByPath(inboxPath))
+                    .isEqualTo(inbox);
+                softly.assertThat(testee.findMailboxWithPathLike(inboxSearchQuery))
+                    .containsOnly(inbox);
+            }));
+        }
+
+        @Disabled("JAMES-3056 find by renamed name returns unexpected results")
```

Comment:
Can you try to fix this in another PR?

Reply:
I don't have a clear way to fix this issue, by undo what had been done?

Reply:
Why do we read acl at the mapper level upon rename? 

Is it really necessary or is there some uneeded calls harming consistency hidding under the hood?

I bet we rely on retrieveMailbox which also calls the acl while reading mailboxDao directly would prevent us from doing the (potentially failing uneeded) acl call.

That's why I'm asking for what looks like a easy fix ;-)

Reply:
See #3107